website/source/guides/stateful-workloads/host-volumes.md

```diff
@@ -0,0 +1,338 @@
+---
+layout: "guides"
+page_title: "Stateful Workloads with Nomad Host Volumes"
+sidebar_current: "guides-stateful-workloads"
+description: |-
+  There are multiple approaches to deploying stateful applications in Nomad.
+  This guide uses Nomad Host Volumes deploy a MySQL database.
+---
+
+# Stateful Workloads with Nomad Host Volumes
+
+Nomad Host Volumes can manage storage for stateful workloads running inside your
+Nomad cluster. This guide walks you through deploying a MySQL workload to a node
+containing supporting storage.
+
+## Reference Material
+
+- [Nomad `host_volume` specification](/docs/configuration/client.html#host_volume-stanza)
+- [Nomad `volume` specification](/docs/job-specification/volume.html)
+- [Nomad `volume_mount` specification](/docs/job-specification/volume_mount.html)
+
+## Estimated Time to Complete
+
+20 minutes
+
+## Challenge
+
+Deploy a MySQL database that needs to be able to persist data without using
+Docker volumes.
+
+## Solution
+
+Configure Nomad Host Volumes on a Nomad client node in order to persist data
+in the event that the container is restarted.
+
+## Prerequisites
+
+To perform the tasks described in this guide, you need to have a Nomad
+environment with Consul installed. You can use this [project][repo] to easily
+provision a sandbox environment. This guide will assume a cluster with one
+server node and three client nodes.
+
+~> **Please Note:** This guide is for demo purposes and is only using a single
+server node. In a production cluster, 3 or 5 server nodes are recommended.
+
+### Prerequisite 1: Install the MySQL client
+
+We will use the MySQL client to connect to our MySQL database and verify our data.
+Ensure it is installed on a node with access to port 3306 on your Nomad clients:
+
+Ubuntu:
+
+```bash
+$ sudo apt install mysql-client
+```
+
+CentOS:
+
+```bash
+$ sudo yum install mysql
+```
+
+macOS via Homebrew:
+
+```bash
+$ brew install mysql-client
+```
+
+### Step 1: Create a Directory to Use as a Mount Target
+
+On a Nomad client node in your cluster, create a directory that will be used for
+persisting the MySQL data.  For this example, let's create the directory
+`/opt/mysql/data`.
+
+```bash
+sudo mkdir -p /opt/mysql/data
+```
+
+You might need to change the owner on this folder if the Nomad client does not
+run as the `root` user.
+
+```bash
+sudo chown «Nomad user» /opt/mysql/data
+```
+
+
+### Step 2: Configure the `mysql` Host Volume on the Client
+
+Edit the Nomad configuration on this Nomad client to create the Host Volume.
+
+Add the following to the `client` stanza of your Nomad configuration:
+
+```hcl
+  host_volume "mysql" {
+    path="/data/mysql"
+    read_only="false"
+  }
+```
+
+Save this change, and then restart the Nomad service on this client to make the
+Host Volume active.
```

Comment:
Potentially include a "You can verify this by running `nomad node status -short <node-id>`"?